hudi-common/src/main/java/org/apache/hudi/common/model/HoodieIndexDefinition.java

```diff
@@ -145,20 +165,43 @@ public static Builder newBuilder() {
     return new Builder();
   }
 
+  /**
+   * Create a new Builder pre-populated with values from this instance.
+   */
+  public Builder toBuilder() {
+    Builder builder = new Builder();
+    builder.withIndexName(this.indexName)
+        .withIndexType(this.indexType)
+        .withIndexFunction(this.indexFunction)
+        .withSourceFields(new ArrayList<>(this.sourceFields))
+        .withIndexOptions(new HashMap<>(this.indexOptions));
+    if (this.version != null) {
+      builder.withVersion(this.version);
+    }
+    return builder;
+  }
+
   public static class Builder {
 
+    // e.g. create index <user index name> on myTable using column_stats(ts) options(expr='from_unixtime', format='yyyy-MM-dd')
```

Comment:
do we need this.. its a good comment. but feels like it should be in the actual field in IndexDefintion. not the builder